libs/inc/vdb.hpp

```diff
@@ -264,9 +322,42 @@ namespace VDB {
                 VCursorRelease( curs );
                 throw Error(rc, __FILE__, __LINE__);
             }
-            return Cursor(const_cast<VCursor *>(curs), n);
+            return Cursor(const_cast<VCursor *>(curs), columns);
+        }
+
+        typedef std::vector< std::string > ColumnNames;
+        ColumnNames physicalColumns() const 
```

Comment:
What is the intended use of these names? I ask because not all physical columns are also readable. For example, `ALT_READ` is a physical column but you can't read it directly on its own via cursor open. You may need a function for readable columns, or something more, like a function that returns something more complicated, like an array of structs with info about every column it can "see", e.g.
{
    std::string name;
    bool isReadable, isPhysical, isVirtual;
}